src/contrib/mlir/core/pass/ng_dialect_builder.cpp

```diff
@@ -323,96 +318,128 @@ void NgDialectConversionPass::buildNgDialect(mlir::FuncOp function)
 }
 
 template <>
-mlir::Operation* NgDialectConversionPass::COMPILE_OP_DECL(ngraph::op::Add)
+mlir::Operation*
+    NgDialectConversionPass::createOp<ngraph::op::Add>(NgDialectConversionPass& NgDialectObj,
```

Comment:
Why do you need to pass a reference to an NGDialectConversionPass here?

Reply:
This is the signature of the function as defined as COMPILE_OP_DECL. I don't like the macro function signature hiding since it makes the code hard to read so I just expanded the macro to the explicit function call. The signature is unchanged.